drivers/include/net/netdev/ieee802154.h

```diff
@@ -108,6 +108,16 @@ typedef struct {
     uint8_t chan;                           /**< channel */
     uint16_t flags;                         /**< flags as defined above */
     /** @} */
+#ifdef MODULE_GNRC_NETIF2_DEDUP_BCAST
+    struct {
+        uint8_t dst[IEEE802154_LONG_ADDRESS_LEN];   /**< destination address */
```

Comment:
I don't see any reason why you should need to store the destination information. Each sender should not reuse sequence numbers except for retransmissions (and after 256 packets when the counter rolls over)